src/classes/RD2_ScheduleService.cls

```diff
@@ -0,0 +1,120 @@
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Class with methods to create, modify and visualize RecurringDonationSchedule__c records.
+*
+*/
+
+public with sharing class RD2_ScheduleService {
+
+    private npe03__Recurring_Donation__c parentRecurringDonation;
+    @TestVisible
+    private List<RecurringDonationSchedule__c> recurringDonationSchedules = new List<RecurringDonationSchedule__c>();
+
+    private void init(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        parentRecurringDonation = rd;
+        if (rdSchedules != null) {
+            recurringDonationSchedules = rdSchedules;
+        }
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd) {
+        init(rd, null);
+    }
+
+    public RD2_ScheduleService(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdSchedules) {
+        init(rd, rdSchedules);
+    }
+
+    /*******************************************************************************************************
+    * @description Initializes RecurringDonationSchedule__c record(s) associated with a parent
+    * npe03__Recurring_Donation__c record (parentRecurringDonation)
+    * @return List<RecurringDonationSchedule__c>
+    */
+    public List<RecurringDonationSchedule__c> initializeSchedule() {
+        if (parentRecurringDonation != null && parentRecurringDonation.Id != null && parentRecurringDonation.Status__c != 'Closed') {
+            recurringDonationSchedules.add(new RecurringDonationSchedule__c());
+            if (parentRecurringDonation.Day_of_Month__c == '1st and 15th' && parentRecurringDonation.npe03__Installment_Period__c == 'Monthly') {
+                recurringDonationSchedules.add(new RecurringDonationSchedule__c());
+            }
+        }
+        generateRecurringDonationScheduleDetails();
+        return recurringDonationSchedules;
+    }
+
+    // TODO: Update schedule to reflect parentRecurringDonation - W-036113
+    public List<RecurringDonationSchedule__c> updateSchedule() {
+        return null;
+    }
+
+    // TODO: Will return next Donation date - W-036251
+    public Date getNextDonationDate() {
+        return null;
+    }
+
+    // TODO: Will return an inner class (TBD)
+    public List<Date> visualizeSchedule() {
+        return null;
+    }
+
+    private void generateRecurringDonationScheduleDetails() {
+        Integer recordCount = 1;
+        for (RecurringDonationSchedule__c rds : recurringDonationSchedules) {
+            rds.InstallmentAmount__c = parentRecurringDonation.npe03__Installment_Amount__c;
```

Comment:
I recommend moving everything in this `if` block into a separate method so that there just a single line for something like `buildScheduleRecord(parentRecurringDonation);`. Plus that will allow the method to be used by other methods as we eventually build out the updateSchedules logic.